src/components/profile-page/settings/profile-display-name.tsx

```diff
@@ -26,32 +30,30 @@ export const ProfileDisplayName: React.FC = () => {
     }
   }, [userName])
 
-  if (!userName) {
-    return <Alert variant={'danger'}>User not logged in</Alert>
-  }
-
-  const changeNameField = (event: ChangeEvent<HTMLInputElement>) => {
-    setSubmittable(!regexInvalidDisplayName.test(event.target.value))
+  const onChangeDisplayName = useCallback((event: ChangeEvent<HTMLInputElement>) => {
+    setSubmittable(!REGEX_INVALID_DISPLAY_NAME.test(event.target.value))
     setDisplayName(event.target.value)
-  }
+  }, [])
 
-  const doAsyncChange = async () => {
-    await updateDisplayName(displayName)
-    await fetchAndSetUser()
-  }
-
-  const changeNameSubmit = (event: FormEvent) => {
-    doAsyncChange().catch(() => setError(true))
-    event.preventDefault()
-  }
+  const onSubmitNameChange = useCallback(
+    (event: FormEvent) => {
+      event.preventDefault()
+      updateDisplayName(displayName)
+        .then(async () => {
+          await fetchAndSetUser()
+        })
```

Comment:
No need for that. You can just return the Promise from fetchAndSetUser

Reply:
I just copy-and-pasted this snippet from the unrefactored code. I probably should have looked at it.